### PR TITLE
Mavlink Fixes for NXPhlite 

### PR DESCRIPTION
**Pull request by davids5**
@dagar - Fixes syntax error and sets correct ports for mavlink.

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d/rc.mavlink [hunk 1]
```diff
@@ -82,132 +84,135 @@ fi
...
 
-if param compare SYS_COMPANION 20
-then
-	syslink start
-	mavlink start -d /dev/bridge0 -b 57600 -m osd -r 40000
-fi
+	if param compare SYS_COMPANION 20
+	then
+		syslink start
+		mavlink start -d /dev/bridge0 -b 57600 -m osd -r 40000
```
Comment (dagar) [inline ROMFS/px4fmu_common/init.d/rc.mavlink]: This one isn't dependant on MAVLINK_COMPANION_DEVICE.

Comment (davids5) [reply ROMFS/px4fmu_common/init.d/rc.mavlink]: I do not follow the intent of this part of the RC well. 

Do we need to discriminate? - The default is not 'none'  - how many knobs should we have and what do they do? 

SYS_COMPANION is a 'select one mode' Knob.
MAVLINK_COMPANION_DEVICE is the Knob that selects the device
Do we add HAS_MAVLINK_COMPANION?

Options:
param set SYS_COMPANION -1 in NXPhlite
Leave MAVLINK_COMPANION_DEVICE as none and let mavling fail to open it. (does mavlink fail well?)

What else?

How do you see this best resolved?

Comment (davids5) [reply ROMFS/px4fmu_common/init.d/rc.mavlink]: @dagar - I split them out into dependant on MAVLINK_COMPANION_DEVICE

**Comment by dagar** [comment]
@PX4/testflights could you please do a short test flight with this PR on an nxphlite + any random pixhawk variant?

Make sure the telemetry radios work (as well as can be expected).

**Comment by dagar** [comment]
Reinitialize as in it resets? Can you enable logging from boot and try again? It sounds like it's hard faulting. Probably a separate issue, but we might as well start debugging it here.

**Comment by davids5** [comment]
@dagar @Tony3dr - I hit the same bug. Please see if the last commit fixes it for you as well.

**Comment by Avysuarez** [comment]
@davids5 already work it!

**Comment by davids5** [comment]
@Avysuarez does the board have the "U50" rework?

**Comment by Avysuarez** [comment]
@davids5 Yes, the telemetry was working.

**Comment by davids5** [comment]
@Tony3dr, @Avysuarez @dagar - I found the regression effecting all the Serial IO. (QGC, SBUS), we were missing the back ports that went upstream prior to the DMA work.

**Comment by davids5** [comment]
@dagar How is this for a Plan 

1) DONE Get PX4 nuttx branch with backpurts merged.
2) DONE update the px4 submodule here in and force push.
3) @PX4/testflights does a santy test.
4) Merge to master and rebase https://github.com/PX4/Firmware/pull/10303, then @PX4/testflights can see if the sensor issues are resolved or if we need more.

**Comment by dagar** [comment]
Sounds good to me.

**Comment by LorenzMeier** [comment]
@davids5 Please rebase, thanks!

**Comment by davids5** [comment]
@PX4/testflights - I have tested this one my HW. Except for as noted below. The coms are working.

 I either have an issue with the GPS/MAG of there is a bug I need you to confirm if you are seeing the same thing. 

1) Build and upload the firmware with QGC or `make nxphlite-v3_default upload`
  (do not connect debugger or console)
2) Connect buzzer.
3) Power from battery cycle the power and see if the boot hangs.
Observe the green LED it should be blinking. If it is not or the buzzer is stuck you are seeing what I am.

Thank you!

**Comment by davids5** [comment]
We are going to merge this to master and rebase https://github.com/PX4/Firmware/pull/10303 the re-test.

**Pull request merged**